specs/proposals.md

```diff
@@ -35,24 +37,62 @@ are part of later specification milestones.
 
 ## Proposing L2 Output Commitments
 
-The proposer's role is to construct and submit output roots, which are commitments made on a configurable interval,
+The proposer's role is to construct and submit output roots, which are commitments to the L2's state,
 to the `L2OutputOracle` contract running on L1. It does this by running the [L2 output proposer](../op-proposer/),
 a service which periodically queries the rollup node's
 [`optimism_outputAtBlock` rpc method](./rollup-node.md#l2-output-rpc-method) for the latest output root derived
 from the latest [finalized](rollup-node.md#finalization-guarantees) L1 block. The construction of this output root is
 described [below](#l2-output-commitment-construction).
 
-If there is no newly finalized output, the service continues querying until it receives one. It then submits this
-output, and the appropriate timestamp, to the [L2 Output Root](#l2-output-root-smart-contract) contract's
-`proposeL2Output()` function. The timestamp block number must correspond to the `startingBlockNumber` plus the next
-multiple of the `SUBMISSION_INTERVAL` value.
+An implementation may submit output commitments that correspond to unsafe (non-finalized) L2 blocks
+but this comes with risk as it will be possible to submit L2 blocks that do not correspond to the
+output commitments that have already been made available.
 
-The proposer may also delete multiple output roots by calling the `deleteL2Outputs()` function and specifying the
+### L2OutputOracle v1.0.0
+
+The submission of output proposals is permissioned to a single account. It is expected that this
+account continues to submit output proposals over time to ensure that user withdrawals do not halt.
+
+The [L2 output proposer](../op-proposer) is expected to submit output roots on a deterministic
+interval based on the configured `SUBMISSION_INTERVAL` in the `L2OutputOracle`. The larger
+the `SUBMISSION_INTERVAL`, the less often L1 transactions need to be sent to the `L2OutputOracle`
+contract, but L2 users will need to wait a bit longer for an output root to be included in L1 that
+includes their intention to withdrawal from the system.
+
+The honest `op-proposer` algorithm assumes a connection to the `L2OutputOracle` contract to know
+the L2 block number that corresponds to the next output proposal that must be submitted. It also
+assumes a connection to an `op-node` to be able to query the `optimism_syncStatus` RPC endpoint.
+
+```python
```

Comment:
I guess we have already referenced the `optimism_outputAtBlock` API, but IMO the pseudocode is an implementation detail (same with the RPC mention)

Reply:
I'm not sure exactly what you mean here, are you suggesting we delete the pseudocode? I do think that the pseudocode is useful personally as its the definition of the honest proposer. Describing it in code is much more precise than using human language, I'd consider things on top of this definition to be implementation details

Reply:
Generally, I think it'd be easier to maintain these specs without the specific pseudocode implementation and rpc reference. But, I think it's acceptable here since it is specifically pinned to a version of the `L2OutputOracle` (v1.0.0).

Reply:
I did specifically pin the algo to v1 for this purpose

Reply:
Are we good to resolve this and merge?